ui/src/components/pages/Pipeline/partials/Graph/partials/NodeInfo/partials/Pods/partials/PodDetails/partials/PodLogs/index.tsx

```diff
@@ -228,20 +255,30 @@ export function PodLogs({
 
   useEffect(() => {
     if (!search) {
-      setFilteredLogs(logs);
+      if(showPreviousLogs){
+        setFilteredLogs(previousLogs);
+      }else{
+        setFilteredLogs(logs);
+      }
       return;
     }
     const searchLowerCase = search.toLowerCase();
-    const filtered = logs.filter((log) =>
+    const filtered = showPreviousLogs ? previousLogs.filter((log) =>
```

Comment:
we can merge the ternary condition like `(showPreviousLogs ? previousLogs : logs)?.filter...`